### DISCUSSION TITLE
Apply Patterns options INSIDE each Specific Pattern

### DISCUSSION THREAD

**Discussion by FelipeRearden** [jglev/obsidian-apply-patterns-plugin#8] [Ideas]
[More info about me here](https://github.com/publicus/obsidian-apply-patterns-plugin/discussions/3)

Hello @publicus!!!

Would be amazing with ewe could define the pattern option (`whole line` or `to selection`) when I am creating a Pattern.

**From the example in your git:**
- For the Set tasks as Complete Pattern  we could have a selector to choose between `whole line` or `to selection`

![Screen Shot 2021-08-10 at 08 57 01](https://user-images.githubusercontent.com/85180696/128862959-8af7886d-974a-4082-ba0b-15f4f93b2b18.png)

For these Patterns pre-defined patterns, the Apply Pattern prompt (`whole line` or `to selection`) should not appear. 
![Screen Shot 2021-08-10 at 08 48 02](https://user-images.githubusercontent.com/85180696/128863381-b2e15fba-b19e-4c94-b194-c748b94258cf.png)

Let me know if you need more information.

Thanks and have a great day!

**Comment by jglev** [comment]
Thank you for writing!

Just a quick note for anyone else reading here that I've commented on this thread just now [here](https://github.com/publicus/obsidian-apply-patterns-plugin/discussions/5#discussioncomment-1159933), consolidating the two threads into one proposed approach.

**Reply by FelipeRearden** [reply]
Thanks for doing this @publicus. 100% crystal clear for me :)

**Reply by FelipeRearden** [reply]
@publicus 
The new feature is running perfect in 1.2.0 on Obsidian 0.12.10 on MacOS

For me, this issue is **closed**

Thanks !!!
